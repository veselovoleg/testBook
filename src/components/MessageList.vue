<template>
    <ul>
        <Message :message="message" v-for="(message, index) in messages" :key="index"/>
    </ul>
</template>

<script>
import Message from './Message'

export default {
  name: 'list',
  props: ['messages'],
  components: {Message}
}
</script>

<style scoped>
    li{
        list-style: none;
    }
</style>
