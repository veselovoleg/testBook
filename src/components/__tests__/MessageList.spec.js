import { mount } from "@vue/test-utils";
import MessageList from "@/components/MessageList.vue";

describe("MessageList.test.js", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(MessageList, {
            // Be aware that props is overridden using `propsData`
            propsData: {
                messages: ["Cat"]
            }
        });
    });

    it('has received ["Cat"] as the message property', () => {
        expect(wrapper.vm.messages).toEqual(["Cat"]);
    });
    it("has the expected html structure", () => {
        expect(wrapper.element).toMatchSnapshot();
    });
});