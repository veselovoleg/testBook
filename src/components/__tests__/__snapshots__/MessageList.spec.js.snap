// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList.test.js has the expected html structure 1`] = `
<ul>
  <li
    class="message"
  >
    Cat
  </li>
</ul>
`;
